app/services/background.py

```
@@ -343,23 +394,58 @@ async def _autonomous_cycles_service(self):
+                    await asyncio.wait_for(
+                        master_controller.run_global_autonomous_cycle(),
+                        timeout=300  # 5 minute timeout
+                    )
                     
                 except ImportError:
                     self.logger.warning("Master controller not available for autonomous cycles")
+                    await asyncio.sleep(300)  # Wait 5 minutes before retrying
+                    continue
+                    
+                except asyncio.TimeoutError:
+                    self.logger.error("Autonomous cycle timed out - this indicates a hanging operation")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)